dashboard/test/models/queued_account_purge_test.rb

```diff
@@ -0,0 +1,27 @@
+require 'test_helper'
+
+class QueuedAccountPurgeTest < ActiveSupport::TestCase
+  test "requires a user" do
+    assert_raises ActiveRecord::StatementInvalid do
+      QueuedAccountPurge.create(reason_for_review: 'test')
+    end
+  end
+
+  test "resolve! purges the user account" do
+    qap = create :queued_account_purge
+    user = qap.user
+
+    AccountPurger.any_instance.expects(:purge_data_for_account).with(user).once
+    qap.resolve!
+  end
+
+  test "resolve! deletes the QueuedAccountPurge" do
```

Comment:
Should we also have a test for when `resolve!` still fails to purge the user - to verify the `QueuedAccountPurge` record is not destroyed?

Reply:
Definitely should, good catch.